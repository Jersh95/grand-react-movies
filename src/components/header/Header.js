import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import {BrowserRouter, Link, Switch, Route} from 'react-router-dom';
import './header.scss';
import CatalogContainer from '../catalog/CatalogContainer';

export default class Header extends React.Component {

  state = {

  };

  render() {
    return (
      <div id='menu-router'>
        <BrowserRouter>
          <Navbar bg="dark" variant="dark" expand='lg'>
            <Navbar.Brand>
              <Link to='/'>Grand React Movies</Link>
            </Navbar.Brand>
            <Navbar.Toggle aria-controls='basic-navbar-nav' />
            <Navbar.Collapse>
              <Nav id='header-nav' className='mr-auto'>
                <Link to='/shop'>Shop</Link>
              </Nav>
            </Navbar.Collapse>
          </Navbar>

          <div>
            <Switch>
              <Route exact path='/shop'>
                <CatalogContainer/>
              </Route>
            </Switch>
          </div>
        </BrowserRouter>
      </div>
    );
  }

}