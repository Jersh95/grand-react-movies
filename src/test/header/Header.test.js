import React from 'react';
import renderer from 'react-test-renderer';
import Header from "../../components/header/Header";

test('Header renders', () => {
  const component = renderer.create(<Header/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});