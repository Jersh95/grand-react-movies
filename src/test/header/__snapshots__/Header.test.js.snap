// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  id="menu-router"
>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Grand React Movies
      </a>
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr-auto navbar-nav"
        id="header-nav"
        onKeyDown={[Function]}
      >
        <a
          href="/shop"
          onClick={[Function]}
        >
          Shop
        </a>
      </div>
    </div>
  </nav>
  <div />
</div>
`;
